### PR TITLE
Implement AddHostPerLine function

### PR DESCRIPTION
**Pull request by praveenkumar**
This will try to solve the issue which windows user
is having where they can't have more than 9 hosts for single
IP.

- https://github.com/goodhosts/hostsfile/issues/18
- https://github.com/code-ready/crc/issues/2710

This patch will not break existing Add function so the consumer
of this library have a way to consume either of this function as
per need.

### CODE REVIEW

**Comment by gbraad** [comment]
While 'insane', according to https://docs.microsoft.com/en-us/previous-versions//bb727005(v=technet.10)#the-hosts-file the hostname itself can only be 25 characters in length on Windows. Is this something we wanna check or ignore?

**Comment by cfergeau** [comment]
>  can only be 25 characters in length

25 -> 255  (25 seemed really short :)

**Comment by cfergeau** [comment]
@luthermonson any thoughts on this PR? We've been having multiple reports from our users of too long hosts file lines on Windows. I'd actually expect `Add` to split the lines if/when needed, which does not seem to be happening? This new `AddHostsPerLine` could work as an alternative.

**Comment by cfergeau** [comment]
> I'd actually expect Add to split the lines if/when needed, which does not seem to be happening?

Actually this seems to be what https://github.com/goodhosts/hostsfile/pull/17#pullrequestreview-728061761 was about.
After just that change, `Add` behaves as expected wrt line splitting:
```
diff --git a/const.go b/const.go
index 75b5d69..05413a0 100644
--- a/const.go
+++ b/const.go
@@ -3,7 +3,7 @@
 package hostsfile
 
 const (
-	HostsPerLine  = -1 // unlimited
+	HostsPerLine  = 4
 	HostsFilePath = "/etc/hosts"
 	eol           = "\n"
 )
diff --git a/hosts.go b/hosts.go
index 54471ac..6486b21 100644
--- a/hosts.go
+++ b/hosts.go
@@ -124,6 +124,7 @@ func (h *Hosts) Add(ip string, hosts ...string) error {
 		}
 		h.Lines[position].Hosts = hostsCopy
 		h.Lines[position].Raw = h.Lines[position].ToRaw() // reset raw
+		h.HostsPerLine(HostsPerLine)
 	}
 
 	return nil
```

**Comment by gbraad** [comment]
@luthermonson Any thoughts?

**Comment by tomjn** [comment]
@luthermonson there' 2 PR's trying to fix this bug and a 3rd that had a question that went unanswered. Is everything ok? Do you need to deputize more maintainers/testers?

Over at VVV we setup various roles so that contributors had some access to allow approvals/assignments, as well as merge rules so they could merge once an approval from someone else happened, all without giving them the keys to the kingdom

**Comment by luthermonson** [comment]
im not sure i like how everyone is trying to solve this problem. in #17 HostsPerLine was a cleanup function and running every Add seems counter productive to just running it at the end. And making an AddHostPerLine func sort of ruins the point of abstracting the datastructure behind the veil in the package as oppose to running HostsPerLine(1) during cleanup. Should we just do a cli flag of --hosts-per-line so the user can do --hosts-per-line=1 and let that be set at call time?

File: hosts.go [hunk 1]
```diff
@@ -129,6 +129,23 @@ func (h *Hosts) Add(ip string, hosts ...string) error {
...
 }
 
+// AddHostPerLine adds entry of the host in separate line.
+func (h *Hosts) AddHostPerLine(ip string, hosts ...string) error {
+	if net.ParseIP(ip) == nil {
+		return fmt.Errorf("%q is an invalid IP address", ip)
+	}
+
+	for _, host := range hosts {
+		// ip not already in hostsfile
```
Comment (luthermonson) [inline hosts.go:139]: dont make a position check, no need to copy/pasta this in

**Comment by luthermonson** [comment]
@praveenkumar we're going to close in favor of #26 which solves the hostsperline issue and adds some more validation and testing

for @gbraad @cfergeau there is now a DNS validator on the Add func so we have some validation about 255 chars and valid dns structure which might fix some issues.. we will merge and pin this in cli

**Pull request closed without merging**